src/test/java/StringManipulationTest.java

```diff
@@ -88,24 +110,41 @@ public void testGeSubStrings1() {
     }
 
     @Test
-    public void testGeSubStrings2() {
-        fail("Not yet implemented");
+    public void testGetSubStrings2() {
+        manipulatedstring.setString("I love programming");
+        String[] subStrings = manipulatedstring.getSubStrings(1, 2);
+
+        assertEquals(subStrings[0],"love");
+        assertEquals(subStrings[1],"programming");
+    }
+    @Test
+    public void testGetSubStrings3() {
+        manipulatedstring.setString("");
+        String[] subStrings = manipulatedstring.getSubStrings(0, 1);
+        assertEquals(0, subStrings.length);
     }
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World!");
+        String[] subStrings = manipulatedstring.getSubStrings(0, 1);
+
+        assertEquals(subStrings[0],"Hello," );
+        assertEquals(subStrings[1],"World!" );
     }
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I am amazing");
+        String[] subStrings = manipulatedstring.getSubStrings(0, 0);
+
+        assertEquals(subStrings[0],"I");
     }
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
-    }
-    @Test
-    public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Java is a programming language");
+        String[] subStrings = manipulatedstring.getSubStrings(4, 6);
+
+        assertEquals("programming", subStrings[0]);
+        assertEquals("language", subStrings[1]);
     }
 
```

Comment:
Another test case you could write would be to have the substring you are trying to retrieve be larger than the length of words in the string. Such as manipulatedstring.getSubstring(122,2000); Then use assertThrows to throw an indexOutOfBounds Exception.